### PR TITLE
LPS-147060 Update blade command to generate rest modules templates. Also, update to use release.product.api

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket -> [LPS-147060](https://issues.liferay.com/browse/LPS-147060)
I send you this PR to check if there are some additional steps to update this blade command.
Thanks for your help 🙂

### CODE REVIEW

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-impl/build.gradle [hunk 1]
```diff
@@ -24,7 +25,10 @@ dependencies {
 	compileOnly group: "org.osgi", name: "org.osgi.service.component.annotations"
 	compileOnly group: "org.osgi", name: "org.osgi.core"
 
-	restBuilder group: "com.liferay", name: "com.liferay.portal.tools.rest.builder", version: "@com.liferay.portal.tools.rest.builder.version@"
+	restBuilder group: "com.liferay", name: "com.liferay.portal.tools.rest.builder"
```
Comment (lawrence-lee) [inline modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/__artifactId__-impl/build.gradle:28]: this dependency isnt managed by our BOMs and Target Platform. anything com.liferay.portal.tools.* isn't managed. We will need to specify this version.

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle [hunk 2]
```diff
@@ -1,5 +1,12 @@
 subprojects {
 	configurations.all {
-		resolutionStrategy.force 'com.liferay:com.liferay.portal.vulcan.api:@com.liferay.portal.vulcan.api.version@'
+	#if (${liferayVersion.startsWith("7.3")})
+	resolutionStrategy.force 'com.liferay:com.liferay.portal.vulcan.api:7.19.2'
+	#elseif(${liferayVersion.startsWith("7.2")} || ${liferayVersion.startsWith("7.1")})
+	resolutionStrategy.force 'com.liferay:com.liferay.portal.vulcan.api:6.9.0'
+	resolutionStrategy.force 'com.liferay:com.liferay.portal.tools.rest.builder:1.0.93'
+	#else
+	resolutionStrategy.force 'com.liferay:com.liferay.portal.vulcan.api:@com.liferay.portal.vulcan.api.version@'
```
Comment (lawrence-lee) [inline modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle:9]: Which versions are you trying to target with this `#else` statement? Right now it's going to capture 7.0 and 7.4 projects.

**Comment by lawrence-lee** [comment]
Thanks for sending this @LuismiBarcos, please see my comments above. Let me know if you have any questions

**Comment by LuismiBarcos** [comment]
Thanks for the review @lawrence-lee I put some comments just to clarify some points. If it is necessary to change something, let me know and I will update it.
Thanks again for your help 🙂

File: modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle (hunk 2 above)
Comment (lawrence-lee) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle:9]: I have a last question, do you need to force the `com.liferay:com.liferay.portal.vulcan.api` version? Could we delegate to the version to the BOM?

Comment (lawrence-lee) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle:8]: @LuismiBarcos ^

Comment (LuismiBarcos) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle:8]: Sorry for the late response.
I'm afraid that we can not delegate the version to the BOM due to the compatibility between Vulcan and REST Builder.
Some REST Builder versions need a specific version of Vulcan-API. Maybe a solution could be to delegate the version of the REST Builder in the BOM? 🤔

Comment (lawrence-lee) [reply modules/sdk/project-templates/project-templates-rest-builder/src/main/resources/archetype-resources/build.gradle:8]: It's not really a build dependency so i'm not sure its worth pursuing at the moment. I just wanted clarity on this. But let me make some changes and I'll send it back to you to look over.

**Pull request closed without merging**